src/utils/handleServerError.ts

```diff
@@ -0,0 +1,26 @@
+import { ResponseError } from '@generated/api/src';
+
+/**
+ * handles server error generated by sdk (Response Error) and sets error state
+ * @param error could be unknown error or ResponseError
+ * @param setError callback to set error state
+ */
+export const handleServerError = async (error: any, setError: (error: string) => void) => {
+  if (error instanceof ResponseError) {
+    const { status, statusText } = error.response;
+
+    if (status === 500) {
+      console.error('Internal Server Error (500):', statusText);
+    } else if (status === 400) {
+      console.error('Bad Request (400):', statusText);
+    } else {
+      console.error(`Error (${status}):`, statusText);
+    }
+    const errorBody = await error.response.json(); // If the response body contains error details
+    const errorMsg = errorBody?.causes?.join('\n') || '[ERROR] Creating installation failed.';
```

Comment:
can you add a comment with a link to https://github.com/amp-labs/openapi/blob/3bc3ab75c3071763e1117f697be3e0fcb636972c/problem/problem.yaml#L85 and explain that all errors returned by the Ampersand API conform to this format?

Reply:
added in part 3. https://github.com/amp-labs/react/pull/547